core/src/test/scala/org/apache/spark/storage/BlockManagerDecommissionIntegrationSuite.scala

```diff
@@ -186,6 +186,8 @@ class BlockManagerDecommissionIntegrationSuite extends SparkFunSuite with LocalS
     sc.addSparkListener(new SparkListener {
       override def onExecutorRemoved(execRemoved: SparkListenerExecutorRemoved): Unit = {
         executorRemovedSem.release()
+        assert(execRemoved.reason == ExecutorDecommission.msgPrefix + "test msg 0" ||
+          execRemoved.reason == "Command exited with code 0")
```

Comment:
It seems like it's not always asserted on `execRemoved.reason == ExecutorDecommission.msgPrefix + "test msg 0"`?

Reply:
Yes. In this test suite `SparkListenerExecutorRemoved`s added here: https://github.com/apache/spark/pull/34536 will be posted too with reason "Command exited with code 0".

Reply:
Shall we do something like:

```scala
var execRemovedEvent = null
override def onExecutorRemoved(execRemoved: SparkListenerExecutorRemoved): Unit = {
  executorRemovedSem.release()
  if (execRemoved.reason == ExecutorDecommission.msgPrefix + "test msg 0") {
    execRemovedEvent = execRemoved
  }
}
...
assert(execRemovedEvent.reason == ExecutorDecommission.msgPrefix + "test msg 0")
```
to ensure we always asserted "ExecutorDecommission.msgPrefix + "test msg 0""?